HACKING.md

```diff
@@ -0,0 +1,409 @@
+then you should be able to find some issues that are good for getting started
+contributing to Salt. [Documentation
+issues](https://github.com/saltstack/salt/issues?q=is%3Aissue+is%3Aopen+label%3Adocumentation+)
+are also good starter issues. When you find an issue that catches your eye (or
+one of your own), it's a good idea to comment on the issue and mention that
+you're working on it. Good communication is key to collaboration - so if you
+don't have time to complete work on the issue, just leaving some information
+about when you expect to pick things up again is a great idea!
+
+## Hacking Away
+
+### Salt, Tests, Documentation, and You
+
+To merge code contributions, Salt requires:
+
+- documentation
+- meaningful passing tests
+- correct code
+
+Documentation fixes just require correct documentation.
+
+#### What If I Don't Write Tests or Docs?
+
+If you aren't into writing documentation or tests, we still welcome your
+contributions! But your PR will be labeled `Needs Testcase` and `Help Wanted`
+until someone can get to write the tests/documentation. Of course, if you have
+a desire but just lack the skill we are more than happy to collaborate and help
+out! There's the [documentation working
+group](https://github.com/saltstack/docs-hub) and the [testing working
+group](https://github.com/saltstack/community/tree/master/working_groups/wg-Testing).
+We also regularly stream our test clinic [live on
+Twitch](https://www.twitch.tv/saltstackinc) every Tuesday afternoon and
+Thursday morning, Central Time. If you'd like specific help with tests, bring
+them to the clinic.  If no community members need help, you can also just watch
+tests written in real time.
+
+### Documentation
+
+Salt uses both docstrings, as well as normal reStructuredText files in the
+`salt/doc` folder for documentation. Since we use nox, you can build your docs
+and view them in your browser with this one-liner:
+
+    python -m nox -e 'docs-html(compress=False, clean=False)'; cd doc/_build/html; python -m webbrowser http://localhost:8000/contents.html; python -m http.server
+
+The first time this will take a while because there are a *lot* of modules.
+Maybe you should go grab some dessert if you already finished that sandwich.
+But once Sphinx is done building the docs, python should launch your default
+browser with the URL http://localhost:8000/contents.html. Now you can navigate
+to your docs and ensure your changes exist. If you make changes, you can simply
+run this:
+
+    cd -; python -m nox -e 'docs-html(compress=False, clean=False)'; cd doc/_build/html; python -m http.server
+
+And then refresh your browser to get your updated docs. This one should be
+quite a bit faster since Sphinx won't need to rebuild everything.
+
+If your change is a doc-only change, you can go ahead and commit/push your code
+and open a PR. Otherwise you'll want to write some tests and code.
```

Comment:
**Question:** How can they mark that it's a docs-only change, perhaps in their MR title?

Reply:
Yeah, I'll call that out explicitly :+1: